plugins/autodim.koplugin/main.lua

```diff
@@ -0,0 +1,257 @@
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                        extra_text = _("Disable"),
+                        extra_callback = function()
+                            self.autodim_starttime_m = -1
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", -1)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(idle_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Dimmer duration: %1 s"), self.autodim_duration_s)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local dimmer_dialog = SpinWidget:new{
+                        title_text = _("Automatic dimmer duration"),
+                        info_text = _("Enter the duration until reaching the final brightness."),
+                        value = self.autodim_duration_s,
+                        default_value = DEFAULT_AUTODIM_DURATION_S,
+                        value_min = 0,
+                        value_max = 300,
+                        value_step = 1,
+                        value_hold_step = 10,
+                        precision = "%1d",
+                        unit = _("s"),
+                        callback = function(spin)
+                            if not spin then return end
+                            self.autodim_duration_s = spin.value
+                            G_reader_settings:saveSetting("autodim_duration_seconds", spin.value)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(dimmer_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Fraction in %1 % of reading brightness"), self.autodim_fraction)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local percentage_dialog = SpinWidget:new{
+                        title_text = _("Fraction of normal brightness"),
+                        info_text = _("The fraction of normal brightness."),
```

Comment:
That is hard to guess what it does ? Is it the final/target dimmed brightness as a fraction of the initial brightness?

Reply:
That is indeed what we've been trying to clarify in the last few rounds of review ^^.

Reply:
Then "The _final_ fraction of normal brightness" would be perfect.